frontend/src/metabase/ui/components/overlays/Modal/index.tsx

```diff
@@ -15,11 +16,14 @@ export const Modal = (props: ModalProps) => {
   );
 };
 
-const ModalRoot = (props: ModalRootProps) => (
-  <PreventEagerPortal>
-    <MantineModal.Root {...props} />
-  </PreventEagerPortal>
-);
+const ModalRoot = (props: ModalRootProps) => {
+  const { withinPortal, ...rootProps } = props;
+  return (
+    <PreventEagerPortal {...props}>
+      <MantineModal.Root {...rootProps} />
```

Comment:
can you leave a little comment here about what you're doing? it took me longer than i care to admit to realize these two components are getting different props for some reason 😅